src/controllers/trustpub/tokens/json.rs

```diff
@@ -0,0 +1,13 @@
+use axum::Json;
+use axum::extract::FromRequest;
+
+#[derive(Debug, Deserialize, FromRequest, utoipa::ToSchema)]
+#[from_request(via(Json))]
+pub struct ExchangeRequest {
+    pub jwt: String,
+}
```

Comment:
We probably want to expand and adjust this to be inline with [RFC 8693](https://datatracker.ietf.org/doc/html/rfc8693#name-request).

The `jwt` here in the original implementation will be the `subject_token` in the spec.

Here's what I'd expect an example token exchange request to look like (a lot of implementations also support JSON content types here too):
```
POST /api/v1/trusted_publishing/tokens
Content-Type: application/x-www-form-urlencoded

grant_type=urn:ietf:params:oauth:grant-type:token-exchange&
subject_token=eyJhbGciOiJSUzI1NiIs...github-oidc-id-token...&
subject_token_type=urn:ietf:params:oauth:token-type:id_token&
audience=crates.io&
requested_token_type=urn:ietf:params:oauth:token-type:access_token
```

Reply:
Resolving this. See https://github.com/rust-lang/crates.io/pull/11131#discussion_r2080144926.